Library/Homebrew/extend/ENV/super.rb

```diff
@@ -72,10 +72,7 @@ def setup_build_environment(formula = nil)
       self["HOMEBREW_FORMULA_PREFIX"] = formula.prefix
     end
 
-    if MacOS::Xcode.without_clt? || (MacOS::Xcode.installed? && MacOS::Xcode.version.to_i >= 7)
-      self["MACOSX_DEPLOYMENT_TARGET"] = MacOS.version.to_s
-      self["SDKROOT"] = MacOS.sdk_path
-    end
+    self["MACOSX_DEPLOYMENT_TARGET"] = MacOS.version.to_s if MacOS::Xcode.installed?
```

Comment:
Could those be considered special cases and treated individually? I just want to avoid replacing a good default with something that appeases a few broken build systems, but creates more harm than good. (As far as I'm aware `MACOSX_DEPLOYMENT_TARGET` is just a different way of passing the `-mmacosx-version-min=<version>` compiler and `-macosx_version_min <version>` linker flag and should otherwise be completely transparent to the build system.)